File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if (prices.length === 0) {
        return 0;
    }

    let min = prices[0];
    let profit = 0;

    for (price of prices) {
        if (price < min) {
            min = price;
        } else if (profit < price - min) {
            profit = price - min;
        }
    }
    
    return profit;
};

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        max_profit = float('-inf')
        min_buy = float('inf')

        for price in prices:
            max_profit = max(max_profit, price - min_buy)
            min_buy = min(min_buy, price)
        
        return max_profit if max_profit > 0 else 0
